### PR TITLE
Add support for dynamics to MS Contactor

### PR DESCRIPTION
**Pull request by andrewlee94**
## Summary/Motivation:
This add support for dynamics to the MS Contactor model.

## Changes proposed in this PR:
- Add method to construct volume and volume fraction terms.
- Add phase fractions to handle multiphase streams
- Added code to construct holdup and accumulation terms as required
- Added accumulation terms to balance equations
- Added units to 0 term in balances for steady-state cases

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by agarciadiego — APPROVED** [review]
LGTM

**Review by ArkoprabhoDasgupta — APPROVED** [review]

File: idaes/models/unit_models/mscontactor.py [hunk 1]
```diff
@@ -1246,11 +1394,78 @@ def _pressure_balance_rule(blk, t, s, stream, uom):
...
+        sum_of_phase_fractions = Constraint(
+            b.flowsheet().time,
+            b.elements,
+            rule=partial(
+                _sum_phase_frac_rule, phase_frac=phase_fraction, phase_list=phase_list
+            ),
+            doc=f"Sum of phase fractions constraint for stream {stream}",
+        )
+        b.add_component(stream + "_sum_phase_fractions", sum_of_phase_fractions)
+
```
Comment (dallan-keylogic) [inline idaes/models/unit_models/mscontactor.py:1453]: Phase volume fraction needs to be given by the property package, and appropriate constraints need to be written here. 

```
def phase_volume_frac_rule(b, p):
    return b.flow_mol_phase[p] * b.dens_mol_phase[p] / sum( b.flow_mol_phase[q] * b.dens_mol_phase[q] for q in b.params.phase_set)
```

File: idaes/models/unit_models/mscontactor.py [hunk 2]
```diff
@@ -558,15 +569,49 @@ def _build_heterogeneous_reaction_blocks(self):
...
+    def _add_geometry(self, uom):
+        if self.config.has_holdup:
+            # Add volume for each element
+            # TODO: Assuming constant volume for now
+            self.volume = Var(
+                self.elements,
+                initialize=1,
+                units=uom.VOLUME,
+                doc="Volume of element",
+            )
```
Comment (dallan-keylogic) [inline idaes/models/unit_models/mscontactor.py:581]: Consider if you want to index `self.volume` by time now if you have plans to make it time-varying eventually. Having an index that a variable is constant along is easier than trying to add the index after the fact.

**Review by dallan-keylogic — CHANGES_REQUESTED** [review]
I think partitioning volume between different streams is fine, with the proviso that specific submodels need to provide the constraints that perform that partitioning. However, the phase volume fraction needs to be provided by property packages, and that constraint should be added as part of this model.

File: idaes/models/unit_models/mscontactor.py (hunk 1 above)
Comment (dallan-keylogic) [reply idaes/models/unit_models/mscontactor.py:1453]: Actually, I got mixed up, it should be 
````
def phase_volume_frac_rule(b, p):
    return b.flow_mol_phase[p] * b.vol_mol_phase[p] / sum( b.flow_mol_phase[q] * b.vol_mol_phase[q] for q in b.params.phase_set)
```

**Review by dallan-keylogic — APPROVED** [review]
Upon further reflection, phase fraction is handled appropriately now, its calculation is outsourced to child models of this class, like other unit models involving the `ControlVolume`.

**Pull request merged**